build.gradle.kts

```diff
@@ -14,13 +14,182 @@
+
+            delete(".env", ".dockerignore", "Dockerfile")
+        }
+
+tasks.clean { dependsOn(removeTempDockerFilesTask) }
+
+var updateDockerEnvTask =
+        tasks.register<Exec>("updateDockerEnv") {
+            description =
+                    "Creates the .env file in the docker folder that contains environment variables for docker"
+            group = "docker"
+
+            workingDir("./docker")
+            commandLine("./update-env.sh", project.version)
+        }
+
+tasks.register<Exec>("createDockerImage") {
+    description = "Creates the docker image of the services based on the current version"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("./docker-build.sh", project.version)
+    finalizedBy(removeTempDockerFilesTask)
+}
+
+tasks.register<Exec>("startDockerContainers") {
+    description = "Starts docker containers of the services based on the current version"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("docker-compose", "up")
+}
+
+tasks.register<Exec>("stopDockerContainers") {
+    description = "Stops running docker containers of the services"
+    group = "docker"
+
+    dependsOn(updateDockerEnvTask)
+    workingDir("./docker")
+    commandLine("docker-compose", "stop")
 }
+
+var buildPrometheusImageTask =
+        tasks.register<DockerBuildImage>("buildPrometheusImage") {
+            description = "Builds Prometheus Docker image"
+            group = "docker intern"
+
+            inputDir.set(file("docker/prometheus"))
+            images.add("swirlds/platform-prometheus:latest")
+        }
+
+var buildGrafanaImageTask =
+        tasks.register<DockerBuildImage>("buildGrafanaImage") {
+            description = "Builds Grafana Docker image"
+            group = "docker intern"
+
+            inputDir.set(file("docker/grafana"))
+            images.add("swirlds/services-grafana:latest")
+        }
+
+var createPrometheusContainerTask =
+        tasks.register<DockerCreateContainer>("createPrometheusContainer") {
+            description = "Create Prometheus Docker container"
+            group = "docker intern"
+            dependsOn(buildPrometheusImageTask)
+
+            targetImageId(buildPrometheusImageTask.get().imageId)
+            hostConfig.portBindings.set(listOf("9090:9090"))
+            hostConfig.autoRemove.set(true)
+        }
+
+var createGrafanaContainerTask =
+        tasks.register<DockerCreateContainer>("createGrafanaContainer") {
+            description = "Create Grafana Docker container"
+            group = "docker intern"
+            dependsOn(buildGrafanaImageTask)
+
+            targetImageId(buildGrafanaImageTask.get().imageId)
+            hostConfig.portBindings.set(listOf("3000:3000"))
+            hostConfig.autoRemove.set(true)
+        }
+
+var stopPrometheusContainer =
+        tasks.register<DockerStopContainer>("stopPrometheusContainer") {
```

Comment:
Adding @nathanklick to this comment.

From my point of view it is not a good idea to use the `com.bmuschko.docker-remote-api` plugin this way. One of the main goals of the plugin is to start/stop containers for tests. This works really good especially by passing the IDs (like `containerId`) to the next task. Since the use case we have here will not be executed in 1 gradle execution the workflow just does not work. I often had the case in that I was not able to stop a container since the containerId was missing / wrong. From my point of view we should at least rewrite the stop command by using exec and executing a shell command that find all running containers for the given name of the container and stop + delete them.

Reply:
We should just use named containers which will resolve these concerns by start/stop/destroy based on container name. 

We should avoid Exec whenever possible because it doesnt scale to windows and linux environments.